lib/DxilPIXPasses/DxilPIXAddTidToAmplificationShaderPayload.cpp

```diff
@@ -45,143 +45,154 @@ void DxilPIXAddTidToAmplificationShaderPayload::applyOptions(PassOptions O) {
-        expanded = ExpandStructType(Ctx, OriginalPayloadStructType);
-      }
-    }
-  }
-
-  AllocaInst *OldStructAlloca = nullptr;
-  AllocaInst *NewStructAlloca = nullptr;
-  std::vector<AllocaInst *> allocasOfPayloadType;
-  for (inst_iterator I = inst_begin(entryFunction), E = inst_end(entryFunction);
-       I != E; ++I) {
-    auto *Inst = &*I;
-    if (llvm::isa<AllocaInst>(Inst)) {
-      auto *Alloca = llvm::cast<AllocaInst>(Inst);
-      if (Alloca->getType() == OriginalPayloadStructPointerType) {
-        allocasOfPayloadType.push_back(Alloca);
-      }
+    if (hlsl::OP::IsDxilOpFuncCallInst(&*I, hlsl::OP::OpCode::DispatchMesh)) {
+      DxilInst_DispatchMesh DispatchMesh(&*I);
+      Type *OriginalPayloadStructPointerType =
+          DispatchMesh.get_payload()->getType();
+      Type *OriginalPayloadStructType =
+          OriginalPayloadStructPointerType->getPointerElementType();
+      ExpandedStruct expanded =
+          ExpandStructType(Ctx, OriginalPayloadStructType);
+
+      llvm::IRBuilder<> B(&*I);
+
+      auto *NewStructAlloca =
+          B.CreateAlloca(expanded.ExpandedPayloadStructType,
+                         HlslOP->GetU32Const(1), "NewPayload");
+      NewStructAlloca->setAlignment(4);
+      auto PayloadType =
+          llvm::dyn_cast<PointerType>(DispatchMesh.get_payload()->getType());
+      SmallVector<Value *, 16> GEPIndices;
+      GEPIndices.push_back(B.getInt32(0));
+      CopyAggregate(B, PayloadType->getPointerElementType(),
+                    DispatchMesh.get_payload(), NewStructAlloca, GEPIndices);
+
+      Constant *Zero32Arg = HlslOP->GetU32Const(0);
+      Constant *One32Arg = HlslOP->GetU32Const(1);
+      Constant *Two32Arg = HlslOP->GetU32Const(2);
+
+      auto GroupIdFunc =
+          HlslOP->GetOpFunc(DXIL::OpCode::GroupId, Type::getInt32Ty(Ctx));
+      Constant *GroupIdOpcode =
+          HlslOP->GetU32Const((unsigned)DXIL::OpCode::GroupId);
+      auto *GroupIdX =
+          B.CreateCall(GroupIdFunc, {GroupIdOpcode, Zero32Arg}, "GroupIdX");
+      auto *GroupIdY =
+          B.CreateCall(GroupIdFunc, {GroupIdOpcode, One32Arg}, "GroupIdY");
+      auto *GroupIdZ =
+          B.CreateCall(GroupIdFunc, {GroupIdOpcode, Two32Arg}, "GroupIdZ");
+
+      // FlatGroupID = z + y*numZ + x*numY*numZ
+      // Where x,y,z are the group ID components, and numZ and numY are the
+      // corresponding AS group-count arguments to the DispatchMesh Direct3D API
+      auto *GroupYxNumZ = B.CreateMul(
+          GroupIdY, HlslOP->GetU32Const(m_DispatchArgumentZ), "GroupYxNumZ");
+      auto *FlatGroupNumZY =
+          B.CreateAdd(GroupIdZ, GroupYxNumZ, "FlatGroupNumZY");
+      auto *GroupXxNumYZ = B.CreateMul(
+          GroupIdX,
+          HlslOP->GetU32Const(m_DispatchArgumentY * m_DispatchArgumentZ),
+          "GroupXxNumYZ");
+      auto *FlatGroupID =
+          B.CreateAdd(GroupXxNumYZ, FlatGroupNumZY, "FlatGroFlatGroupIDupNum");
```

Comment:
"FlatGroFlatGroupIDupNum"
I think this might be a typo

Reply:
Woops, thanks! Looks like I pasted in a new name but didn't select the old one first, resulting in the new name inside the old!
I tend not to enforce names for DXIL values because they aren't actually stable anyway. In fact, the tests tend to go out of their way to match a regex to the actual name...